drivers/dma/dma_xilinx_axi_dma.c

```diff
@@ -0,0 +1,1020 @@
+	barrier_dmem_fence_full();
+
+	if (halted) {
+		uint32_t new_control = 0;
+
+		new_control |= XILINX_AXI_DMA_REGS_DMACR_RS;
+		/* no reset */
+		new_control &= ~XILINX_AXI_DMA_REGS_DMACR_RESET;
+		/* TODO make this a DT parameter */
+		/* for Eth DMA, this should never be used */
+		new_control &= ~XILINX_AXI_DMA_REGS_DMACR_KEYHOLE;
+		/* no cyclic mode - we use completed bit to control which */
+		/* transfers where completed */
+		new_control &= ~XILINX_AXI_DMA_REGS_DMACR_CYC_BD_EN;
+		/* we want interrupts on complete */
+		new_control |= XILINX_AXI_DMA_REGS_DMACR_IOC_IRQEN;
+		/* we do NOT want timeout IRQs */
+		new_control &= ~XILINX_AXI_DMA_REGS_DMACR_DLY_IRQEN;
+		/* we want IRQs on error */
+		new_control |= XILINX_AXI_DMA_REGS_DMACR_ERR_IRQEN;
+		/* interrupt after every completed transfer */
+		new_control |= 0x01 << XILINX_AXI_DMA_REGS_DMACR_IRQTHRESH_SHIFT_BITS;
+		/* no timeout */
+		new_control |= 0x0 << XILINX_AXI_DMA_REGS_DMACR_IRQDELAY_SHIFT_BITS;
+
+		LOG_DBG("New DMACR value: %" PRIx32, new_control);
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Waddress-of-packed-member"
+		dma_xilinx_axi_dma_write_reg(&channel_data->channel_regs->dmacr, new_control);
+		/* need to make sure start was committed before writing tail */
+		barrier_dmem_fence_full();
+	}
+
+#ifdef CONFIG_DMA_64BIT
+	dma_xilinx_axi_dma_write_reg(&channel_data->channel_regs->taildesc,
+				     (uint32_t)(((uintptr_t)current_descriptor) & 0xffffffff));
+	dma_xilinx_axi_dma_write_reg(&channel_data->channel_regs->taildesc_msb,
+				     (uint32_t)(((uintptr_t)current_descriptor) >> 32));
+#else
+	dma_xilinx_axi_dma_write_reg(&channel_data->channel_regs->taildesc,
+				     (uint32_t)(uintptr_t)current_descriptor);
+#endif
+#pragma GCC diagnostic pop
+
+	dma_xilinx_axi_dma_enable_cache();
+
+	irq_unlock(irq_key);
+
+	/* commit stores before returning to caller */
+	barrier_dmem_fence_full();
+
+	return 0;
+}
+
+static int dma_xilinx_axi_dma_stop(const struct device *dev, uint32_t channel)
+{
+	const struct dma_xilinx_axi_dma_config *cfg = dev->config;
+	struct dma_xilinx_axi_dma_data *data = dev->data;
+	struct dma_xilinx_axi_dma_channel *channel_data = &data->channels[channel];
+
+	uint32_t new_control;
+
+	if (channel >= cfg->channels) {
+		LOG_ERR("Invalid channel %" PRIu32 " - must be < %" PRIu32 "!", channel,
+			cfg->channels);
+		k_oops();
+		return -EINVAL;
+	}
+
+	new_control = channel_data->channel_regs->dmacr;
+	/* RS = 0 --> DMA will complete ongoing transactions and then go into hold */
+	new_control = new_control & ~XILINX_AXI_DMA_REGS_DMACR_RS;
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Waddress-of-packed-member"
+	dma_xilinx_axi_dma_write_reg(&channel_data->channel_regs->dmacr, new_control);
+#pragma GCC diagnostic pop
+
+	/* commit before returning to caller */
+	barrier_dmem_fence_full();
+
+	return 0;
+}
+
+static int dma_xilinx_axi_dma_suspend(const struct device *dev, uint32_t channel)
```

Comment:
If this is unimplemented (along with resume) set the function pointers to NULL in the API struct instead, this is dealt with at an API level for you already 

e.g. https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/dma.h#L501

Reply:
Done